splunk_eventgen/eventgen_core.py

```diff
@@ -548,8 +548,7 @@ def join_process(self):
         :return:
         '''
         try:
-            while not self.sampleQueue.empty() or self.sampleQueue.unfinished_tasks > 0 or not self.workerQueue.empty(
-            ) or self.workerQueue.unfinished_tasks > 0:
+            while not self.sampleQueue.empty() or self.sampleQueue.unfinished_tasks > 0 or not self.workerQueue.empty():
```

Comment:
in threaded mode, I am concerned that the jobs might end prematurely. How about implement a check to see if the mode is threading and check if there are unfinished tasks?